include/libopencm3/stm32/g0/rcc.h

```diff
@@ -0,0 +1,948 @@
+/** @defgroup rcc_defines RCC Defines
+ *
+ * @ingroup STM32G0xx_defines
+ *
+ * @brief <b>Defined Constants and Types for the STM32G0xx Reset and Clock Control</b>
+ *
+ * @version 1.0.0
+ *
+ * LGPL License Terms @ref lgpl_license
+ *  */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**@{*/
+
+#ifndef LIBOPENCM3_RCC_H
+#define LIBOPENCM3_RCC_H
+
+#include <libopencm3/stm32/pwr.h>
+
+/* --- RCC registers ------------------------------------------------------- */
+
+#define RCC_CR_OFFSET			0x00
+#define RCC_CR					MMIO32(RCC_BASE + RCC_CR_OFFSET)
+#define RCC_ICSCR_OFFSET		0x04
+#define RCC_ICSCR				MMIO32(RCC_BASE + RCC_ICSCR_OFFSET)
+#define RCC_CFGR_OFFSET			0x08
+#define RCC_CFGR				MMIO32(RCC_BASE + RCC_CFGR_OFFSET)
```

Comment:
why this style?  it's not done anywhere else, and I can't say I like it.

Reply:
Ah! I attempted to do some clever stuff to to reduce amount of somewhat more or less duplicated code in rcc between devices, but nothing came out nicely and forgot about them.. I'll get rid of these offset defines.

Reply:
hmm. or not. it's been a while now - it's also done like in include/libopencm3/stm32/l4/rcc.h, but i used it for all the definitions. I'll rest

Reply:
right, it's only for the ones that are used in the bit macros, so just for periphheral enable/reset...

the ```	RST_GPIOE = _REG_BIT(RCC_AHB2RSTR_OFFSET, 4),``` sort of stuff.